developer_onboarding_guide.md

```diff
@@ -0,0 +1,35 @@
+# Developer Onboarding Guide
+
+## Prerequisites
+- [go](https://go.dev/doc/install) (version > 1.18)
+- [gofumpt](https://github.com/mvdan/gofumpt)
+- [golangci-lint](https://golangci-lint.run/) (version 1.48.0)
+
+## Creating New PR and Getting It Ready to Be Merged to the Main Branch
+
+To start with, you need to create a new branch where you'll develop the new features. Generally, we use a branch naming convention that follows the format:  `<your_name>/<feature_description>`.
+
+Once your branch is pushed to GitHub, you should open a Draft Pull Request. This way, other team members can review your code and approve it upon completion.
```

Comment:
Draft - only if you need some preliminary opinion about the approach, or you want to run the tests.

Reply:
Fixed.